### PR TITLE
key-auth+cli e2e test connect #345

### PR DESCRIPTION
**Pull request by DrMegavolt**
connect #345

### CODE REVIEW

File: test/common/gateway.helper.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        reject(err);
+      });
+    }
+  });
+};
+
+module.exports.startGatewayInstance = function ({dirInfo, gatewayConfig}) {
+  return this.findOpenPortNumbers(4)
+      .then(ports => {
+        console.log(ports);
```
Comment (kevinswiber) [inline test/common/gateway.helper.js]: Should be able to get rid of the console.log statement.

File: test/common/gateway.helper.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          let gatewayProcess = fork(modulePath, [], {
+            cwd: dirInfo.basePath,
+            env: childEnv
+          });
+
+          gatewayProcess.on('error', err => {
+            reject(err);
+          });
+          let count = 0;
+          let interval = setInterval(() => {
```
Comment (kevinswiber) [inline test/common/gateway.helper.js]: :+1:

**Comment by kevinswiber** [comment]
@DrMegavolt Can you refactor the OAuth2 Authorization Code test to take advantage of the new helpers you built and perhaps throw in the retry counter you added?

**Comment by DrMegavolt** [comment]
@kevinswiber auth-code and hot-reload now uses refactored `find ports` method.
Create Gateway is bit different in each case, probably my helper will need revision once we have more use cases, for now I left original code

**Pull request merged**